boxsdk/object/user.py

```diff
@@ -1,11 +1,45 @@
 # coding: utf-8
 
 from __future__ import unicode_literals
+from functools import partial
 
 from .base_object import BaseObject
+from .group_membership import GroupMembership
 
 
 class User(BaseObject):
     """Represents a Box user."""
 
     _item_type = 'user'
+
+    def groups(self, starting_index=0, limit=100, include_page_info=False):
```

Comment:
This is called `groups`, but it returns `GroupMembership`s, not `Group`s.

Idea: rename this method to `membership` (then it mirrors `Group.membership()`). Optionally, add another method

```python
def groups(self, starting_index=0, limit=100, include_page_info=False):
    for group_membership, current_page_size, index_in_current_page in self.membership(starting_index=starting_index, limit=limit, include_page_info=True):
        group = group_membership.group
        if include_page_info:
            yield group, current_page_size, index_in_current_page
        else:
            yield group
```